### PR TITLE
Add file staging queue

### LINKED ISSUE #26: Download outputfiles in background
Given a job is completed on a remote system without a shared file system. 
The output files of the job must be copied back to the file system which can be seen by the bartender service.

In #17 the copying of output files is done during handling of `GET /api/job/<jobid>` or `GET /api/job` HTTP requests.
This works fine for small amounts (<1Mb) of bytes to copy, but for big amounts (~40Gb) this will cause timeouts on the HTTP requests.

It would be nice if we could use https://fastapi.tiangolo.com/tutorial/background-tasks or use some file copying queue.

A new job state should be introduced like `staging_out` to signal that copying of files is happening for a job.
The `staging_out` job state can also be used to prevent copying multiple times This is not a big issue as files should not change after job completion.

For uploading input files to a remote file system the background task is already used.

### PR DESCRIPTION
**Pull request by sverhoeven**
Fixes #26 

To test
1. Configure bartender to use a isolated filesystem, for example by configuring the Slurm Docker container at https://github.com/i-VRESSE/bartender/blob/9085050b52e08de7ba266979540af51bb68fc655/config-example.yaml#L34-L52
2. Submit job as https://github.com/i-VRESSE/bartender/tree/staging-queue-26#word-count-example
3. If `GET /api/job/{jobid}/stdout` route returns something then this is proof that the output files where copied from the container.

The job state returned by `GET /api/job/{jobid}` can be `staging_out`, which signifies that web service is busy copying files back.

### CODE REVIEW

File: src/bartender/web/api/job/views.py [hunk 1]
```diff
@@ -101,18 +122,10 @@ async def retrieve_job_stdout(
...
     )
-    if job.state not in {"ok", "error"}:
+    if job.state not in CompletedStates:
         raise HTTPException(
             status_code=status.HTTP_425_TOO_EARLY,
             detail="Stdout not ready. Job has not completed.",
         )
     stdout: Path = context.job_root_dir / str(jobid) / "stdout.txt"
     return FileResponse(stdout)
-
```
Comment (sverhoeven) [inline src/bartender/web/api/job/views.py:111]: Moved to #37

File: src/bartender/web/api/job/views.py [hunk 2]
```diff
@@ -101,18 +122,10 @@ async def retrieve_job_stdout(
...
-
-# TODO add job deletion route.
-# Should job be removed from db?
-# Decide what to do in following cases:
-# * when job queued
-# * when job is running
-# * when job is completed
```
Comment (sverhoeven) [inline src/bartender/web/api/job/views.py:118]: Moved to https://github.com/i-VRESSE/bartender/issues/37

**Review by Peter9192 — APPROVED** [review]
Looks good, works well! You already answered my questions offline :-)

**Comment by sverhoeven** [comment]
Offiline we talked about enqueuing a function or id in queue. See https://github.com/i-VRESSE/bartender/blob/ca0d520c802238487719dcb3a9a05a496dbe2f71/src/bartender/web/api/job/sync.py#L180-L183
I have replaced the enqueuing of a function with `(job.id, state and destination_name)`.

**Pull request merged**